File: Assignment_26/program5.c
///////////////////////////////////////////////////////////////////////
//
// Name :          program5.c
// Discription :   Reverse the String
// Input :         String
// Output :        String 
// Author :        Shivam Rajendra Kale
// Date :          09/06/2025
//
////////////////////////////////////////////////////////////////////////

/*
  Start
    Accept String from user 
      reverse the string
  Stop    
*/


#include<stdio.h>

void strReverse(char *str)
{
   int iCnt = 0;
   char *start = NULL;
   char *end = NULL;
   char temp = '\0';

   if(NULL == str)
   {
      return ;
   }

   start = str;
   end = str;

   while(*end != '\0')
   {
      end++;
   }
   end--;

   while(start < end)
   {
      temp = *start;
      *start = *end;
      *end = temp;

      start++;
      end--;
   }
  
}

int main()
{
   char Arr[20] = {'\0'};

   printf("Enter string :\n");
   scanf("%[^'\n']s",Arr);

   strReverse(Arr); 

   printf("Updated String is :%s",Arr);

   return 0;

}

File: Assignment_53/program5.java
//////////////////////////////////////////////////////////////////////////
//
// Name :          program5.java
// Discription :   Check wether Array is Pallindrome or not
// Input :         Integer
// Output :        Integer
// Author :        Shivam Rajendra Kale
// Date :          17/10/2025
//
//////////////////////////////////////////////////////////////////////////

import java.util.*;

class MyArray
{
    public boolean ChkPallindrome(int Arr[])
    {
        int i = 0, j = 0;
        boolean bFlag = true;
        
        for(i = 0, j = (Arr.length - 1); i <= j; i++, j--)
        {
            if(Arr[i] != Arr[j])
            {
                bFlag = false;
                return bFlag;
            }
        }

        return bFlag;
    }
}

class program5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        int i = 0;

        System.out.println("Enter isze of first Array : ");
        int iSize = sobj.nextInt();

        int Arr[] = new int [iSize];

        System.out.println("Enter elements : ");
        for(i = 0; i < iSize; i++)
        {
            Arr[i] = sobj.nextInt();
        }

        MyArray mobj = new MyArray();

        boolean bRet = mobj.ChkPallindrome(Arr);

        if(bRet == true)
        {
            System.out.println("Array is Pallndrome ");
        }
        else 
        {
            System.out.println("Array is not Pallndrome ");
        }

        sobj.close();
    }
}